src/sklearn_raster/estimator.py

```diff
@@ -444,9 +442,170 @@ def kneighbors(
+        check_output_for_nodata: bool = True,
+        **transform_kwargs,
+    ) -> FeatureArrayType:
+        """
+        Apply the transformation to n-dimensional X features.
+
+        Parameters
+        ----------
+        X : Numpy or Xarray features
+            The n-dimensional input features. Array types should be in the shape
+            (features, ...) while xr.Dataset should include features as variables.
+            Features should correspond with those used to fit the estimator.
+        skip_nodata : bool, default=True
+            If True, NoData and NaN values will be skipped during prediction. This
+            speeds up processing of partially masked features, but may be incompatible
+            if estimators expect a consistent number of input samples.
+        nodata_input : float or sequence of floats, optional
+            NoData values other than NaN to mask in the output features. A single value
+            will be broadcast to all features while sequences of values will be assigned
+            feature-wise. If None, values will be inferred if possible based on
+            available metadata.
+        nodata_output : float or int or tuple, optional
+            NoData samples in the input features will be replaced with this value in the
+            output features. If the value does not fit the array dtype returned by the
+            estimator, an error will be raised unless `allow_cast` is True. Defaults to
+            np.nan.
+        ensure_min_samples : int, default 1
+            The minimum number of samples that should be passed to `transform`. If the
+            array is fully masked and `skip_nodata=True`, dummy values (0) will be
+            inserted to ensure this number of samples. The minimum supported number of
+            samples depends on the estimator used. No effect if the array contains
+            enough unmasked samples or if `skip_nodata=False`.
+        allow_cast : bool, default=False
+            If True and the estimator output dtype is incompatible with the chosen
+            `nodata_output` value, the output will be cast to the correct dtype instead
+            of raising an error.
+        check_output_for_nodata : bool, default True
+            If True and `nodata_output` is not np.nan, a warning will be raised if the
+            selected `nodata_output` value is returned by the estimator, as this may
+            indicate a valid sample being masked.
+        **transform_kwargs
+            Additional arguments passed to the estimator's `transform` method.
+
+        Returns
+        -------
+        Numpy or Xarray features
+            The transformed features. Array types will be in the shape (features, ...)
+            while xr.Dataset will store features as variables, with the feature names
+            based on the estimator's `get_feature_names_out` method.
+        """
+        feature_names = self._wrapped.get_feature_names_out()
```

Comment:
Can you just use `self.get_feature_names_out()` as the `AttrWrapper` should resolve it?  I see you using `self._wrapped` in a couple of other places, so it's very possible I'm not understanding a subtlety here.

Reply:
Great question, and you're right that `self.get_feature_names_out()` would correctly access the wrapped instance's method (of course that doesn't apply to methods overridden by `FeatureArrayEstimator` since those supersede the wrapped attributes, e.g. `self.transform` is different than `self._wrapped.transform`).

I think I prefer the explicitness of accessing attributes through `_wrapped` at the expense of some brevity,  just to make it clear at a glance where things are defined and avoid any confusion with overridden methods.

Reply:
> I think I prefer the explicitness of accessing attributes through `_wrapped` at the expense of some brevity, just to make it clear at a glance where things are defined and avoid any confusion with overridden methods.

Yes, that was the subtlety that I was missing - namely, that the wrapper may have methods defined that would supersede the wrapped object.  I completely agree with your logic of being explicit here.